### PR TITLE
Update: Qwen3-14B decode FP32 inter-layer carry + fused layer output

### PR DESCRIPTION
**Pull request by luohuan19**
## Summary
- Carry the inter-layer hidden as FP32 (was BF16), dropping per-layer BF16 rounding for more precise decode numerics (golden argmax is the sanity check, not bit-equality).
- Fuse the layer output and the next layer's x*gamma into one `dcr_xgamma` task (DOWN_ON-way): emit `out` (residual) and `normed_out` (x*gamma) from the same in-register chunk, removing the `out_partial` scratch + extra copy task and the GM round-trip for x_gamma.
- Whole-layer manual scope (x_gamma..rope) with explicit `deps=` edges; cross-iteration TaskId carries (`prev_out_tids` / `prev_normed_tids`) gate the next layer's seeds / QKV. `copy_hidden` casts the external BF16 embed input -> FP32 to seed layer 0; `cast_lmhead_in` casts the final FP32 hidden -> BF16 for the unchanged `rms_lm_head`.
- `QKV_OK` 4 -> 5 (`QKV_K_SLICE=1024`, 1:1 partial-Q split).
- KV cache stays PAGED (`block_table` read / `slot_mapping` write, `SEQ_TILE` == serving page_size 128): the per-layer cache base and block-table row stride are derived from the runtime-dynamic pool dims, so decode reads the same paged pool prefill writes. (An intermediate contiguous-pool detour was reverted within this branch.)
- `--enable-l2-swimlane` becomes an optional integer `PERF_LEVEL` (was a bare boolean flag): bare flag selects level 4 (full); levels 1=AICore timing, 2=+dispatch/fanout, 3=+sched phases, 4=+orch phases; 0 (default) disables. Lets swimlane capture granularity be tuned per run.

## Related Issues
None

### CODE REVIEW

File: models/qwen3/14b/decode_layer.py [hunk 1]
```diff
@@ -1533,12 +1721,14 @@ def stack0(t, reps):  # replicate along dim 0
...
+        # host ref: chain N single-layer decode_fwd_layers (N==1) -> final RMSNorm -> lm_head.
+        # Each step reloads the un-stacked single-layer inputs and overrides the hidden.
+        _CHUNK_NLAYERS = 1
         ref_hidden = inputs[0]; ref_out = None
         for _step in range(N):
             ci = load_inputs(args.data_dir / "in"); ci[0] = ref_hidden
             ref_out = torch.zeros(BATCH, HIDDEN, dtype=torch.bfloat16)
-            qwen3_decode_mpmd(*ci, ref_out, config=run_cfg)
+            decode_fwd_layers(*ci, ref_out, config=run_cfg)
             ref_hidden = ref_out.clone()
```
[the remark this answers is not in the document]
Comment (luohuan19) [reply models/qwen3/14b/decode_layer.py]: Fixed in `f4ffa5c` — the host reference now runs a single N-layer `decode_fwd_layers` chunk (`_CHUNK_NLAYERS = N`), keeping the inter-layer residual FP32 until the chunk-boundary BF16 cast, matching `decode_fwd`'s FP32-carry-until-LM-head path. The E702 semicolons aren't flagged here (ruff selects only `F`/pyflakes), but the restructure removes them anyway.
